### ISSUE TITLE
vagga Disk full

### ISSUE THREAD

**Issue by lclc** [tailhook/vagga-docker#9]
Hi, I don't know if this is related to vagga or vagga-docker, but I somehow managed to fill up my disk completely (in the vagga instance):

root@29b2056b695a:/work# df -BG
Filesystem     1G-blocks  Used Available Use% Mounted on
/dev/vda2            60G   58G        0G 100% /
tmpfs                 2G    0G        2G   0% /dev
shm                   1G    0G        1G   0% /dev/shm
tmpfs                 2G    0G        2G   0% /sys/fs/cgroup
osxfs               465G  132G      333G  29% /work
tmpfs                 1G    0G        1G   0% /tmp
tmpfs                 1G    0G        1G   0% /run
tmpfs                 1G    0G        1G   0% /gopath/pkg
(from a vagga instance doing a 'vagga shell')

To start again over with a modified, clean vagga I just deleted the .vagga folder. Is this the correct way to do this?

I also stopped the vagga docker instances, but the disk is still full (now I can't do 'vagga shell' anymore).

**Comment by tailhook** [comment]
TL;DR: I've just updated vagga version used by vagga-docker (you need to reinstall it from git), now `clean` command should work (it's good idea to occasionally run it):
```
vagga _clean --unused
```

Longer answer:

1. When you're changing `vagga.yaml` another version of container is built. And all of them are kept alive until you run `vagga _clean`. Also, vagga keeps some files in cache dir for faster rebuilds, but it rarely grows too large. There should be no other disk space "leaks" in vagga. In native vagga you can also clean everything by removing `.vagga` (unless you have an external `storage-dir` in settings)

2. When you're using `vagga-docker` all your containers are created inside the volume named `vagga-<something>`. And the name of that volume is stored in `.vagga/.docker-volume`. So removing `.vagga` doesn't help. You need to run `vagga _clean` or if you already removed `.vagga/.docker-volume`. You might use docker's functionality `docker volume rm vagga-<something>` (it should be easy to figure out name of the volume by using `docker volume list`).

3. Because docker is still not native in OS X (unlike a lot of people think) it may or may not free storage on the host system because virtual image may still use some parts of the allocated blocks. But it seems that the issue you have is lack of storage space in the guest system (inside docker) rather than host.

**Comment by lclc** [comment]
Thanks for the fast and detailed answer!

Yes, it was only inside the system. Plenty left on the host.

**Comment by tailhook** [comment]
Looks like the issue may be closed.
